### ISSUE TITLE
close browsed file (aka remove file from history)

### ISSUE THREAD

**Issue by pepdiz** [mittelmark/shtmlview#1]
> I understand now that the close a file feature should remove the file from the history and load the previous file instead. And that the next widgets *browse* command might start a complete new history if requested. So for instance if I call the widget directly from the terminal with:
> 
> $ tclsh shtmlview.tcl file1.html file2.html test.md
> 
> or even
> $ tclsh shtmlview.tcl *.html *.md
> 
> after closing file1.html from the widget therafter file2.html should be displayed and the history should be cleaned, so that I can process all files similar like in an image viewer where I can walk through the images of a directory with a single N key for instance.
> Right?

more or less right, in my idea when you close the file1.html it's not the case that history is cleaned but modified, removing the reference to file1.html in the history.  

you can perform a clean up of the history after closing files, of course, but I prefer just reordering the history

if you have a history list of opened files and each movement backward and forward through the list implies reopen the selected file, simply removing the entry from the history list is enough

but if you keep all opened files in memory then close a file means to remove the file from memory and from the list of opened files, so further moves simply skip the position of the now inexistent file.

in other words, closing a file simply means removing it from the history list.  What "remove" means depends on what you store in the history list (a string with path and name of file, a file handle, a file object...)

**Comment by mittelmark** [comment]
Ok, I see your point. Sounds not to difficult, I will try to merge this functionality with a browse for multiple files

`$widget browse file1 file2 file3`

 should create a history of the three files where with f (forward) and b (backward) I could switch from one file to the other and with k (kill) the current file will be deleted from the history and the next (forward) file would be shown, that way it would be easy browsing a set of HTML and Markdown files simply using the keyboard. Just the filenames btw are stored in the history. So they will re-rendered always. The widget is quite slow but should work for most files sufficiently fast.

**Comment by mittelmark** [comment]
I now uploaded the close file functionality. You can use the close file button or the 'q' key to remove a file from history, the next file in history will be displayed or the last file in the remaining history if the closed file was deleted.
You can now as well browse multiple files, even from the terminal:
tclsh shtmlview.tcl TclCmd/*.htm 
would show the first file and place all other files in history, where you can walk everything with f and b keys.
